runtime/src/accounts_index.rs

```diff
@@ -76,6 +81,17 @@ impl<'a, T: 'a + Clone> AccountsIndex<T> {
         (reclaims, list.is_empty())
     }
 
+    pub fn purge_exact(&self, pubkey: &Pubkey, slots: HashSet<Slot>) -> (SlotList<T>, bool) {
+        let list = &mut self.account_maps.get(&pubkey).unwrap().1.write().unwrap();
+        let reclaims = list
+            .iter()
+            .filter(|(slot, _)| slots.contains(&slot))
+            .cloned()
+            .collect();
+        list.retain(|(slot, _)| !slots.contains(slot));
```

Comment:
nit: you may be able to do this in one loop if you build the `reclaims` within the `retain`

Reply:
hmm, true. I wonder if that has a speed advantage. I'll experiment for a follow-up.